app/main.py

```diff
@@ -2,24 +1,102 @@
+from abc import ABC, abstractmethod
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.protected_name = "_" + name
+
+    def __get__(
+            self,
+            instance: "SlideLimitationValidator",
+            owner: type
+    ) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(
+            self,
+            instance: "SlideLimitationValidator",
+            value: int
+    ) -> None:
+        if not (self.min_amount <= value <= self.max_amount):
+            raise ValueError(f"{self.protected_name} must be between {self.min_amount}"
+                             f" and {self.max_amount}")
+        setattr(instance, self.protected_name, value)
 
 
 class Visitor:
-   pass
+    def __init__(
+            self,
+            name: str,
+            age: int,
+            weight: int,
+            height: int
+    ) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(
+            self,
+            age_range: IntegerRange,
+            weight_range: IntegerRange,
+            height_range: IntegerRange
+    ) -> None:
+        self.age_range = age_range
+        self.weight_range = weight_range
+        self.height_range = height_range
```

Comment:
The `IntegerRange` class is defined with descriptor methods (`__get__`, `__set__`, `__set_name__`), but it is not used as a descriptor in the `SlideLimitationValidator` class. If the intention was to use it for automatic validation, consider using it as a descriptor for the `age_range`, `weight_range`, and `height_range` attributes.